pkg/apis/minio.min.io/v2/helper.go

```diff
@@ -761,6 +761,26 @@ func (t *Tenant) CreateUsers(madmClnt *madmin.AdminClient, userCredentialSecrets
 	return nil
 }
 
+// CheckBucketsExist checks if the given buckets exist in the MinIO server
+func (t *Tenant) CheckBucketsExist(minioClient *minio.Client, buckets ...Bucket) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*20)
+	defer cancel()
+	allBuckets, err := minioClient.ListBuckets(ctx)
+	if err != nil {
+		return err
+	}
+	allBucketsMap := make(map[string]bool)
+	for _, bucket := range allBuckets {
+		allBucketsMap[bucket.Name] = true
+	}
+	for _, bucket := range buckets {
+		if !allBucketsMap[bucket.Name] {
```

Comment:
why can't we use `HeadBucket` here?

Reply:
I can't found that. Is there something I missed?